src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,228 @@
+                cars.add(parseCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get cars by driver with id "
+                    + driverId + " ", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        List<Driver> drivers = new ArrayList<>();
+        String query = "SELECT d.id, d.name, d.license_number "
+                + "FROM cars_drivers cd JOIN drivers d ON cd.driver_id = d.id "
+                + "WHERE car_id = ? AND d.is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversByCarIdStatement
+                        = connection.prepareStatement(query)) {
+            getDriversByCarIdStatement.setLong(1, carId);
+            ResultSet resultSet = getDriversByCarIdStatement.executeQuery();
+            while (resultSet.next()) {
+                drivers.add(parseDriver(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get drivers for car with id "
+                    + carId + " ", e);
+        }
+    }
+
+    private int removeDriversFromCar(Car car) {
+        String deleteCarsDriversQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarsDriversStatement
+                        = connection.prepareStatement(deleteCarsDriversQuery)) {
+            deleteCarsDriversStatement.setLong(1, car.getId());
+            return deleteCarsDriversStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update car drivers for " + car + " ", e);
+        }
+    }
+
+    private int addDriversToCar(Car car) {
+        int addedDrivers = 0;
+        if (car.getDrivers() == null) {
+            return addedDrivers;
+        }
+        for (Driver driver : car.getDrivers()) {
+            String addCarsDriversRelationQuery = "INSERT INTO cars_drivers (car_id, driver_id) "
+                    + "VALUES (?, ?)";
+            try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement addCarsDriversRelationStatement
+                            = connection.prepareStatement(addCarsDriversRelationQuery)) {
+                addCarsDriversRelationStatement.setLong(1, car.getId());
+                addCarsDriversRelationStatement.setLong(2, driver.getId());
+                addedDrivers += addCarsDriversRelationStatement.executeUpdate();
+            } catch (SQLException e) {
+                throw new DataProcessingException("Couldn't add drivers for " + car + " ", e);
+            }
+        }
+        return addedDrivers;
+    }
```

Comment:
We use many connections in this method. Better to use one connection and one preparedStatement example of **for** in connection
 for (Driver driver: car.getDrivers()) {
                insertDriversStatement.setLong(2, driver.getId());
                insertDriversStatement.executeUpdate();
 }

Reply:
Method changed